src/entity/mixins/runtime_items.js

```diff
@@ -145,6 +145,17 @@ export const RuntimeItemsUILogicMixin = (superclass) => {
         getResultByName(name) {
             return this.results.find((r) => r.name === name);
         }
+
+        filterResultsByName(name) {
+            return this.results.filter((r) => r.name === name);
```

Comment:
Sometimes we store strings, not objects, and this logic wouldn't work. Should be `(r || r.name) === name` here and on line 154 and 156

Reply:
Okay, but should we then also update all the other functions in this file? `getResultsByName`, `resultNames`, etc. all expect an object with the attribute `name`.